sds-go/go/scanner.go

```diff
@@ -115,53 +115,64 @@ func (s *Scanner) scanEncodedEvent(encodedEvent []byte) ([]byte, []RuleMatch, er
 	// Meaning that the data in `rv` is a copy owned by Go of what's in rvdata.
 	rv := C.GoBytes(unsafe.Pointer(rvdata), C.int(retsize))
 
-	processed, ruleMatches, err := decodeResponse(rv)
+	mutated, processed, ruleMatches, err := decodeResponse(rv)
 	if err != nil {
-		return nil, nil, fmt.Errorf("scan: %v", err)
+		return false, nil, nil, fmt.Errorf("scan: %v", err)
 	}
 
-	return processed, ruleMatches, nil
+	return mutated, processed, ruleMatches, err
 }
 
 // Scan sends the string event to the SDS shared library for processing.
 // Returned values:
-//   - the processed log if any mutation happened
+//   - whether a mutation happened or not
+//   - the resulting log (possibly be unchanged)
 //   - rules matches if any rule has matched
 //   - a possible error
-func (s *Scanner) Scan(event []byte) ([]byte, []RuleMatch, error) {
+func (s *Scanner) Scan(event []byte) (bool, []byte, []RuleMatch, error) {
```

Comment:
It might be worth returning a new struct here? 4 values is a lot to remember what they are / the order of them. Especially with a `bool` where the meaning might not be very clear.

Reply:
The `[]RuleMatch` also has enough info to determine if the event was mutated or not, but if you think a `bool` is easier that's fine.

Reply:
Hmm you are right about the RuleMatch containing the `MatchingType`... 🤔 but it would mean a `for range` on all matches to see if it has been mutated... 
Good idea I'll return a struct indeed, way more information returned separately here 👍